deepchem/molnet/load_function/delaney_datasets.py

```diff
@@ -3,29 +3,52 @@
 """
 import os
 import logging
-import deepchem
+import deepchem as dc
+from deepchem.molnet.load_function.molnet_loader import _MolnetLoader
+from deepchem.data import Dataset
+from typing import List, Optional, Tuple, Union
 
 logger = logging.getLogger(__name__)
 
-DEFAULT_DIR = deepchem.utils.data_utils.get_data_dir()
 DELANEY_URL = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/delaney-processed.csv"
+DELANEY_TASKS = ['measured log solubility in mols per litre']
 
 
-def load_delaney(featurizer='ECFP',
-                 split='index',
-                 reload=True,
-                 move_mean=True,
-                 data_dir=None,
-                 save_dir=None,
-                 **kwargs):
-  """Load delaney dataset
+class _DelaneyLoader(_MolnetLoader):
 
-  The Delaney(ESOL) dataset a regression dataset containing structures and
+  def create_dataset(self) -> Dataset:
+    logger.info("About to featurize Delaney dataset.")
+    dataset_file = os.path.join(self.data_dir, "delaney-processed.csv")
+    if not os.path.exists(dataset_file):
+      dc.utils.data_utils.download_url(url=DELANEY_URL, dest_dir=self.data_dir)
+    loader = dc.data.CSVLoader(
+        tasks=DELANEY_TASKS, feature_field="smiles", featurizer=self.featurizer)
+    return loader.create_dataset(dataset_file, shard_size=8192)
+
+  def get_transformers(self, dataset: Dataset) -> List[dc.trans.Transformer]:
```

Comment:
To allow users to specify their own transformers, perhaps we can make `get_transformers` be the default if no user-specified transformations are applied.